daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala

```diff
@@ -441,16 +441,38 @@ class IdeClient(val compiledPackages: CompiledPackages) extends ScriptLedgerClie
                   result = Success(Right(results.toSeq))
               }
           }
+        case SResultFinalValue(v) =>
+          // The final result should always be unit.
+          result = Failure(new RuntimeException(s"Unexpected non-unit final result: $v"))
```

Comment:
I guess if we want to go for FATAL, this should also be FATAL?

Reply:
Agreed, done.